src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -60,56 +60,73 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
+                    if (matchingItem != null)
                     {
-                        return matchingDocument;
+                        return matchingItem;
                     }
                 }
                 else
                 {
-                    // We were not passed a project context.  This can happen when the LSP powered NavBar is not enabled.
-                    // This branch should be removed when we're using the LSP based navbar in all scenarios.
-
-                    var solution = documents.First().Project.Solution;
-                    // Lookup which of the linked documents is currently active in the workspace.
-                    var documentIdInCurrentContext = solution.Workspace.GetDocumentIdInCurrentContext(documents.First().Id);
-                    return solution.GetRequiredDocument(documentIdInCurrentContext);
+                    return defaultGetter();
                 }
             }
 
-            // We either have only one document or have multiple, but none of them  matched our context. In the
+            // We either have only one item or have multiple, but none of them  matched our context. In the
             // latter case, we'll just return the first one arbitrarily since this might just be some temporary mis-sync
             // of client and server state.
-            return documents[0];
+            return items[0];
+        }
+
+        public static T FindDocumentInProjectContext<T>(this ImmutableArray<T> documents, TextDocumentIdentifier documentIdentifier, Func<Solution, DocumentId, T> documentGetter) where T : TextDocument
+        {
+            return FindItemInProjectContext(documents, documentIdentifier, projectIdGetter: (item) => item.Project.Id, defaultGetter: () =>
+            {
+                // We were not passed a project context.  This can happen when the LSP powered NavBar is not enabled.
+                // This branch should be removed when we're using the LSP based navbar in all scenarios.
+
+                var solution = documents.First().Project.Solution;
+                // Lookup which of the linked documents is currently active in the workspace.
+                var documentIdInCurrentContext = solution.Workspace.GetDocumentIdInCurrentContext(documents.First().Id);
+                return documentGetter(solution, documentIdInCurrentContext);
+            });
         }
 
         public static Project? GetProject(this Solution solution, TextDocumentIdentifier projectIdentifier)
-            => solution.Projects.Where(project => project.FilePath == projectIdentifier.Uri.LocalPath).SingleOrDefault();
+        {
+            var projects = solution.Projects.Where(project => project.FilePath == projectIdentifier.Uri.LocalPath).ToImmutableArray();
+            return !projects.Any()
+                ? null
+                : FindItemInProjectContext(projects, projectIdentifier, projectIdGetter: (item) => item.Id, defaultGetter: () => projects[0]);
```

Comment:
fix - now we lookup the project file URI in the correct project context (e.g. tfm)